src/server/wesnothd/server.cpp

```diff
@@ -1867,34 +1895,9 @@ void server::handle_player_in_game(player_iterator p, simple_wml::document& data
 	} else if(data.child("stop_updates")) {
 		g.send_data(data, p);
 		return;
-	} else if(simple_wml::node* request = data.child("game_history_request")) {
-		if(user_handler_) {
-			int offset = request->attr("offset").to_int();
-			int player_id = 0;
-
-			// if no search_for attribute -> get the requestor's forum id
-			// if search_for attribute for offline player -> query the forum database for the forum id
-			// if search_for attribute for online player -> get the forum id from wesnothd's player info
-			if(!request->has_attr("search_for")) {
-				player_id = player.config_address()->attr("forum_id").to_int();
-			} else {
-				std::string player_name = request->attr("search_for").to_string();
-				auto player_ptr = player_connections_.get<name_t>().find(player_name);
-				if(player_ptr == player_connections_.get<name_t>().end()) {
-					player_id = user_handler_->get_forum_id(player_name);
-				} else {
-					player_id = player_ptr->info().config_address()->attr("forum_id").to_int();
-				}
-			}
-
-			if(player_id != 0) {
-				LOG_SERVER << "Querying game history requested by player `" << player.name() << "` for player id `" << player_id << "`.";
-				user_handler_->async_get_and_send_game_history(io_service_, *this, p, player_id, offset);
-			}
-		}
-		return;
+	}
 	// Data to ignore.
-	} else if(
```

Comment:
Is there a reason for this formatting change? (While it's just formatting I'd like to keep non-essential changes to a minimum in general. Makes for cleaner history.)

Regarding removing the function from in game handling, would it make sense to handle the request from either lobby or in game? I haven't looked at the client side yet. Perhaps more difficult to support it there but I wouldn't mind if the server supports either.

Reply:
It wasn't intentional, it's just how I happened to cut/paste it to the lobby handler.

Having it available in-game wouldn't be as useful since it would only ever show your own game history since there's not yet a way on the dialog itself to provide a player name. There would also need to be a hotkey and/or menu item to make the request (which doesn't exist yet either) as well as a check to not try to make the request in single player.